### DISCUSSION TITLE
Unable to visualize any output

### DISCUSSION THREAD

**Discussion by staunton-golding** [dipy/dipy#3028] [Scientific Visualization] [answered]
I am following the CSD reconstruction tutorial, but all the plots / images I produce are either all black (with empty pixels / pixel values = 0) or a noisy rainbow, if they are produced at all. I am running the code on a CentOS 7 HPC cluster. 

The window.record (or window.snapshot) line normally causes the code to abort - only not aborting and capturing photos when the code is ran with a remote desktop with the xvfbwrapper wrapper active. These are the black or rainbow photos.

Any help would be greatly greatly appreciated.

**Comment by skoudoro** [comment]
Hi @staunton-golding,

it looks like an OpenGl issue on your CentOs 7 HPC cluster. 

What is you OpenGl version ?  is OpenGl installed ? One way to check OpenGL installation is to check is the following command run correctly inside the hpc terminal :  `glxgears`. (you should visualize a wheel running)

if it is ok. I recommend to define the following environment variable on your HPC: `MESA_GL_VERSION_OVERRIDE=3.3`. 

Let us know if it fixes your issue.

**Comment by alextruong** [comment]
Hello, I've been helping @staunton-golding troubleshoot this issue on our cluster; it's been a bit difficult to sort this out because our system install of OpenGL does not include `swrast_dri.so`, so I've been attempting to get the requisite libraries to be visible from a conda environment. Also, are there any associated options we might be able to specify for `window.*` to use something like nvidia drivers to render instead? I couldn't find any documentation for `viz.window` on the website.

**Reply by skoudoro** [reply]
>  Also, are there any associated options we might be able to specify for window.* to use something like nvidia drivers to render instead?

No, any environment variables come to my mind for that. 

Our visualization depends on VTK. So, it might be possible if you compile VTK locally, on your server. 
Look [here](https://docs.pyvista.org/version/stable/extras/building_vtk#off-screen-plotting-gpu-support) to build VTK, then, our visualisation system will find it and use it. 

I hope it helps.

**Comment by alextruong** [comment, marked as answer]
We ended up having to install `mesa-dri-drivers` to gain access to `swrast_dri.so` in the default location and get things working, but even that required setting `MESA_GL_VERSION_OVERRIDE=3.3` on top of that. Distribution of that package is pending in our environment, but should resolve this discussion at that point. @staunton-golding should be able to confirm at that time.

**Reply by skoudoro** [reply]
thank you for the feedback !
